packaging/macosx/2_build_hb_darktable_default.sh

```diff
@@ -29,5 +29,4 @@ if [[ -d "$buildDir" ]]; then
 fi
 
 # Clean build here
-#../../build.sh --install --build-type Release --prefix "$installDir"
-../../build.sh --install --build-type Release --prefix "$installDir" -- $options
+../../build.sh --install --build-type Release --prefix "$installDir" -- "$options"
```

Comment:
Are you sure, this means that all options as passed to the script as a single parameter as $options is quoted.

Reply:
Quite sure, I tested all variations.
Without the quotes, `build.sh` throws a warning:

`warning: ignoring unknown option -DUSE_IMAGEMAGICK=ON`

Reply:
That’s because you now set USE_GRAPHICSMAGICK to “OFF -DUSE_IMAGEMAGICK”. This is an incorrect fix for sure. Better rework how options is set, these multi-line strings are tricky, maybe try several lines options=“$options -DUSE…” and drop the quotes as suggested by Pascal

Reply:
@parafin: Did you see my last edit (calling `build.sh` from the command line). There is no multi-line string invoked.

Reply:
So there is a bug in `build.sh` I would say. The line taking arguments after `--` is:

```
		--)
			shift
			CMAKE_OPTIONS_FROM_CMDLINE="$@"
			;;
```

And this is pass to the command line later:

`cmd_config="${ASAN_FLAGS}cmake -G \"$BUILD_GENERATOR\" -DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX} -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ${CMAKE_MORE_OPTIONS} ${CMAKE_OPTIONS_FROM_CMDLINE} \"$DT_SRC_DIR\""`

Furthemore if I run your command on GNU/Linux it works:

`./build.sh --install --build-type Release --prefix ../../build/macosx -- -DUSE_GRAPHICSMAGICK=OFF -DUSE_IMAGEMAGICK=ON`

I got the compilation started after full configuration:

```
...
-- The following OPTIONAL packages have not been found:

 * SDL2

-- Configuring done (3.8s)
-- Generating done (2.2s)
-- Build files have been written to: /home/obry/dev/git/darktable/build
[  0%] Built target locale_ca
[  0%] Updating version string (git checkout)
```

So somehting macOS specific, can you try replacing:
`			CMAKE_OPTIONS_FROM_CMDLINE="$@"`
by
`			CMAKE_OPTIONS_FROM_CMDLINE="$*"`
?

Reply:
build.sh is at fault. it continues the cmd arguments loop even after encountering --, while it should break the looping. Also I guess it doesn’t handle arguments with whitespaces, that’s why quoting the variable “fixed” the issue. Better not rely on these quirks, but fix the build.sh.

Reply:
I can reproduce on Windows (bash on UCRT64) as well.

> So somehting macOS specific, can you try replacing:
			CMAKE_OPTIONS_FROM_CMDLINE="$@"
by
			CMAKE_OPTIONS_FROM_CMDLINE="$*"

That doesn't make any difference but inserting a `break` seems to fix it:

```
		--)
			shift
			CMAKE_OPTIONS_FROM_CMDLINE="$@"
			break
			;;
```

Reply:
@TurboGit can you please double check again on Linux? The warning is the very first line of the output and the script continues to build. That's the reason why I didn't noticed that earlier.

Reply:
You're plain right, I missed the first line...

`warning: ignoring unknown option -DUSE_IMAGEMAGICK=ON`

Reply:
I have now removed the quotes again so this PR is only some optical cleanup of the script.

The inserted `break` in `build.sh` fixes the warning on macOS and my Windows setup.
Can you confirm on Linux as well?

Reply:
Ready to go